src/drivers/net/virtio/mod.rs

```diff
@@ -471,20 +528,35 @@ impl VirtioNetDriver {
 	///
 	/// See Virtio specification v1.1. - 3.1.1.
 	///                      and v1.1. - 5.1.5
-	pub fn init_dev(&mut self) -> Result<(), VirtioNetError> {
-		// Reset
-		self.com_cfg.reset_dev();
+	pub fn init_dev(
+		mut com_cfg: ComCfg,
+		notif_cfg: NotifCfg,
+		isr_stat: IsrStatus,
+		mut dev_cfg: NetDevCfg,
+		irq: u8,
+	) -> Result<VirtioNetDriver, VirtioNetError> {
+		let ctrl_vq;
+		let recv_vqs;
+		let send_vqs;
+		let mtu;
+		let mut checksums;
+
+		// We do the whole initialization process inside an inner scope so that if any of the steps fail,
+		// the scope returns an `Err` that we can catch to set the driver status as failed.
+		{
```

Comment:
This does not work. Returning from a scope returns the whole function immediately without inspecting the result. We'd have to use a closure or proper function, since try blocks are not there yet.